src/Microsoft.AspNetCore.Mvc.Core/ForbidObjectResult.cs

```diff
@@ -0,0 +1,128 @@
+        /// specified <paramref name="properties"/>.
+        /// </summary>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <param name="value">Response payload.</param>
+        public ForbidObjectResult(object value, AuthenticationProperties properties)
+            : this(value, Array.Empty<string>(), properties)
+        {
+        }
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbidObjectResult"/> with the
+        /// specified authentication scheme and <paramref name="properties"/>.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication schemes to challenge.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <param name="value">Response payload.</param>
+        public ForbidObjectResult(object value, string authenticationScheme, AuthenticationProperties properties)
+            : this(value, new[] { authenticationScheme }, properties)
+        {
+        }
+        /// <summary>
+        /// Initializes a new instance of <see cref="ForbidObjectResult"/> with the
+        /// specified authentication schemes and <paramref name="properties"/>.
+        /// </summary>
+        /// <param name="authenticationSchemes">The authentication scheme to challenge.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <param name="value">Response payload.</param>
+        public ForbidObjectResult(object value, IList<string> authenticationSchemes, AuthenticationProperties properties)
+            : base(value)
+        {
+            AuthenticationSchemes = authenticationSchemes;
+            Properties = properties;
+        }
+
+        /// <summary>
+        /// Gets or sets the authentication schemes that are challenged.
+        /// </summary>
+        public IList<string> AuthenticationSchemes { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="AuthenticationProperties"/> used to perform the authentication challenge.
+        /// </summary>
+        public AuthenticationProperties Properties { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ExecuteResultAsync(ActionContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger<ForbidResult>();
+
+            logger.ForbidResultExecuting(AuthenticationSchemes);
+
+            if (AuthenticationSchemes != null && AuthenticationSchemes.Count > 0)
```

Comment:
The pattern for doing this is to invoke an executor - see https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs#L36